solutions/advanced/izho-12-xor.mdx

```diff
@@ -0,0 +1,152 @@
+---
+id: izho-12-xor
+source: IZhO 2012
+title: XOR
+author: Nathan Katov
+---
+
+## Explanation
+
+Firstly, the original problem can be simplified slightly. In the question, we are asked to find the longest interval that satisfies this inequality.
+$$
+a_i \oplus a_{i+1} \oplus \dots \oplus a_{i+k-1} \ge x, 1 \le i \le i + k - 1 \le N
+$$
+
+However, if we create a "prefix-xor" array of $a$ then this can be simplified to:
+$$
+p_{i-1} \oplus p_{i+k-1} \ge x, 1 \le i \le i + k - 1 \le N
+$$
+where $p$ is the "prefix-xor" array of $a$.
+
+We will now attempt to solve this problem.
+Firstly, let's iterate over the array from left to right and for every index, $r$, we need to find the smallest index, $l$, such that $ p_{l} \oplus p_{r} \ge x $.
+
+This can be done using a trie. The trie will store the binary representations of the values in the array and the indices of the values. At each node, we should store its two children and the smallest index in the subtree rooted at the node. Using this trie, we will find the values of $p_l$ that work one bit at a time.
+
+To find the possible values of $p_l$, we will traverse through the bits from most significant to least significant. At each bit, we will try both to make this bit on and to make this bit off.
+
+In both cases, we check whether $p_l \oplus p_r = x$ holds at this specific bit. If it does, then this means that this is a possible answer and we must continue traversing (It can be shown that this will only happen for one of the two possible bits). It is also important to note that you must handle the case of total equality carefully.
+
+We also check if $p_l \oplus p_r > x$ holds at this specific bit. If it does, then this means that the xor value is already greater and the less significant bits are irrelevant. So, this is an answer and we take the smallest index in the subtree rooted at this node as all the values in the subtree satisfy the original inequality.
+
+To recap, we will iterate through the array, adding the values into the trie as we go along. At every index, we find the smallest index satisfying the condition and are trivially able to find the longest interval satisfying the original condition.
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N\log10^9)$
```

Comment:
Instead of using 10^9 (a constant value), use a variable value such as a to represent the max possible value in the input array.

Reply:
Fixed